frontend/src/app/person.service.spec.ts

```diff
@@ -1,80 +1,69 @@
+import { TestBed } from '@angular/core/testing';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
 
 import { PersonService } from './person.service';
 import { PersonModel } from './models/person.model';
 
 describe('PersonService', () => {
 
-  beforeEach(() => TestBed.configureTestingModule({
-    providers: [
-      MockBackend,
-      BaseRequestOptions,
-      { provide: Http, useFactory: (backend, options) => new Http(backend, options), deps: [MockBackend, BaseRequestOptions] },
-      PersonService
-    ]
-  }));
-
-  it('should get a person', async(() => {
-    const service: PersonService = TestBed.get(PersonService);
-    const fakePerson: PersonModel = { id: 1 } as PersonModel;
-
-    const mockBackend: MockBackend = TestBed.get(MockBackend);
-    mockBackend.connections.subscribe(connection => {
-      expect(connection.request.url).toBe(`/api/persons/1`);
-      expect(connection.request.method).toBe(RequestMethod.Get);
-      connection.mockRespond(new Response(new ResponseOptions({ body: fakePerson })));
+  let service: PersonService;
+  let http: HttpTestingController;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [ PersonService ],
+      imports: [ HttpClientTestingModule ]
     });
 
-    service.get(1)
-      .subscribe(person => expect(person).toBe(fakePerson));
-  }));
+    service = TestBed.get(PersonService);
+    http = TestBed.get(HttpTestingController);
+  });
+
+  it('should get a person', () => {
+    const expectedPerson: PersonModel = { id: 1 } as PersonModel;
+
+    let actualPerson;
+    service.get(1).subscribe(person => actualPerson = person);
 
-  it('should update a person', async(() => {
-    const service: PersonService = TestBed.get(PersonService);
+    http.expectOne({url: '/api/persons/1', method: 'GET'}).flush(expectedPerson);
+    expect(actualPerson).toEqual(expectedPerson);
+
+  });
+
+  it('should update a person', () => {
     const fakePerson: PersonModel = { id: 2 } as PersonModel;
 
-    const mockBackend: MockBackend = TestBed.get(MockBackend);
-    mockBackend.connections.subscribe(connection => {
-      expect(connection.request.url).toBe(`/api/persons/2`);
-      expect(connection.request.method).toBe(RequestMethod.Put);
-      expect(JSON.parse(connection.request.getBody())).toEqual(fakePerson);
-      connection.mockRespond(new Response(new ResponseOptions({ body: fakePerson })));
-    });
+    service.update(fakePerson).subscribe(() => {});
 
-    service.update(fakePerson)
-      .subscribe(() => {});
-  }));
+    const testRequest = http.expectOne({url: '/api/persons/2', method: 'PUT'});
+    expect(testRequest.request.body).toEqual(fakePerson);
+    testRequest.flush(null);
 
-  it('should create a person', async(() => {
-    const service: PersonService = TestBed.get(PersonService);
-    const fakePerson: PersonModel = { id: 1 } as PersonModel;
+    http.verify();
```

Comment:
Because, if I'm not mistaken, the tests where I used verify() wouldn't fail if I sent a URL to the wrong URL of with the wrong method. Indeed, these requests don't send back anything, and I thus don't have any expectation on the returned value as I have with requests returning something.